### ISSUE TITLE
"odo url create" with "--app" and "--project" flags is broken

### ISSUE THREAD

**Issue by dharmit** [openshift/odo#2204]
Creating a URL using `--app` and `--project` flags from a component directory creates a URL for the component in current working directory.
```sh
$ odo config view
COMPONENT SETTINGS
------------------------------------------------
PARAMETER         CURRENT_VALUE
Type              nodejs
Application       app
Project           myproject
SourceType        local
Ref               
SourceLocation    ./
Ports             8080/TCP
Name              nodejs-nodejs-ex-vaiv
MinMemory         
MaxMemory         
DebugPort         
Ignore            
MinCPU            
MaxCPU 

$ odo url create myurl --app nosuchapp --project nosuchproject
 ✓  URL myurl created for component: nodejs-nodejs-ex-vaiv

To create URL on the OpenShift Cluster, please use `odo push`

$ odo url list
Found the following URLs for component nodejs-nodejs-ex-vaiv in application app:
NAME      STATE          URL     PORT
myurl     Not Pushed     ://     8080

There are local changes. Please run 'odo push'.
$ odo push 
# push was successful and url got created for the component in pwd
```

_Originally posted by @dharmit in https://github.com/openshift/odo/issues/2203#issuecomment-536456785_

**Comment by amitkrout** [comment]
#2247

**Comment by amitkrout** [comment]
closed via #2247 
/close
